### PR TITLE
Update provider statement to use extension in quickstart templates

### PR DESCRIPTION
**Pull request by jason-dou**
`provider` keyword is deprecated in the latest Bicep version. Updating to use `extension`
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoftgraph/msgraph-bicep-types/pull/156)

### CODE REVIEW

File: quickstart-templates/application-serviceprincipal-create-client-resource/main.bicep [hunk 1]
```diff
@@ -1,4 +1,4 @@
-provider microsoftGraph
+extension microsoftGraph
```
Comment (dkershaw10) [inline quickstart-templates/application-serviceprincipal-create-client-resource/main.bicep:1]: For each of the Readmes, do we need to specify the minimum Bicep version number?

**Review by dkershaw10 — DISMISSED** [review]
Left one comment, but this is good to go for me.

File: quickstart-templates/application-serviceprincipal-create-client-resource/main.bicep (hunk 1 above)
Comment (jason-dou) [reply quickstart-templates/application-serviceprincipal-create-client-resource/main.bicep:1]: Yeah we have required Bicep version number in the release notes, but I think it's a good idea to include in each quickstart Readme. I will update it

**Review by dkershaw10 — APPROVED** [review]
LGTM

**Pull request merged**